lib/cc/engine/analyzers/reporter.rb

```diff
@@ -26,12 +30,20 @@ def process_files
             concurrency: engine_config.concurrency
           )
 
+          processed_files_count = 0
+
           pool.run do |file|
+            debug("Processing file: #{file}")
+
             sexp = language_strategy.run(file)
             process_sexp(sexp)
+
+            processed_files_count += 1
```

Comment:
You could also use this class from concurrent-ruby I think http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/AtomicFixnum.html